nixos/modules/services/networking/firewall.nix

```diff
@@ -493,6 +511,80 @@ in
         '';
     };
 
+    networking.firewall.extraAllowed = mkOption {
+      type = types.listOf (types.submodule { options = {
+        sourceAddress = mkOption {
+          type = types.nullOr types.string;
```

Comment:
`string` has merge semantics that may differ from what you expect. If two modules declare an option `foo`, the final value is the concatenation of the two values, so "foo" and "bar" would turn into "foobar" in the final config.